tests/socket.rs

```diff
@@ -568,10 +571,14 @@ fn unix() {
     let addr = SockAddr::unix(path).unwrap();
 
     let listener = Socket::new(Domain::UNIX, Type::STREAM, None).unwrap();
+    #[cfg(target_os = "cygwin")]
+    listener.set_no_peercred().unwrap();
     listener.bind(&addr).unwrap();
     listener.listen(10).unwrap();
```

Comment:
The cygwin thread you shared says that this solution is a "kludge". It sounds like it's not intended that you work around this with `SO_PEERCRED`.

Reply:
Well, it's designed for this case, but not recommended. Should I remove it?

Reply:
Actually I prefer adding this `set_no_peercred`. Many programs, including C & Rust programs, assume the common behavior that `connect()` doesn't block on Unix sockets. This API is designed for the case as a workaround. I think it would be useful to port some Rust programs to Cygwin...